modules/appeals_api/app/models/appeals_api/supplemental_claim.rb

```diff
@@ -342,6 +342,20 @@ def add_error(message)
       errors.add(:base, message)
     end
 
+    # validation (header & body)
+    # Schemas take care of most of the requirements, but we need to check that both header & body data is provided
+    def required_claimant_data_is_present
+      has_claimant_headers = claimant.first_name.present?
+      # form data that includes a claimant is also sufficient to know it's passed the schema
+      has_claimant_data = data_attributes&.fetch('claimant', nil).present?
+
+      return if !has_claimant_headers && !has_claimant_data # No claimant headers or data? not a problem!
+      return if has_claimant_headers && has_claimant_data # Has both claimant headers and data? A-ok!
+
+      add_error('Claimant data was provided but missing claimant headers') unless has_claimant_headers
```

Comment:
@stiehlrod just noting this addition of `add_error`, since I know you're working on the model error side of things in your ticket. This'll need to be updated with whatever changes you end up going with.